core/tests/test-dependencies/root-files-mapping-generator.ts

```diff
@@ -0,0 +1,723 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Script to generate a files to root files mapping for the
+ * Oppia codebase.
+ */
+
+import path from 'path';
+import fs from 'fs';
+import {Decorator, SourceFile, ts} from 'ts-morph';
+import * as cheerio from 'cheerio';
+import {
+  project,
+  ROOT_DIRECTORY,
+  AngularDecorators,
+  getRelativePathToRootDirectory,
+  isNodeModule,
+  getDecorationNodesByTextFromSourceFile,
+  resolveModuleRelativeToRoot,
+  getDecorationNodeText,
+  getValueFromLiteralStringOrBinaryExpression,
+} from './typescript-ast-utilities';
+import {
+  getPageModules,
+  getPageModulesCoveredByRouteMapping,
+} from './route-to-module-mapping-generator';
+
+interface BaseAngularInformation {
+  className: string;
+}
+
+interface AngularModuleInformation extends BaseAngularInformation {
+  type: 'module';
+}
+
+interface AngularComponentInformation extends BaseAngularInformation {
+  type: 'component';
+  selector?: string;
+  templateFilePath?: string;
+}
+
+interface AngularDirectiveOrPipeInformation extends BaseAngularInformation {
+  type: 'directive' | 'pipe';
+  selector?: string;
+}
+
+type AngularInformation =
+  | AngularModuleInformation
+  | AngularComponentInformation
+  | AngularDirectiveOrPipeInformation;
+
+const GIT_IGNORED_EXCLUSIONS = fs
+  .readFileSync(path.resolve(ROOT_DIRECTORY, '.gitignore'), 'utf-8')
+  .split('\n')
+  .filter(line => line.trim() && !line.startsWith('#'));
+
+const FILE_EXCLUSIONS = [
+  ...GIT_IGNORED_EXCLUSIONS,
+  'types',
+  'typings',
+  'scripts',
+  'assets/scripts',
+  'core/tests/build_sources',
+  'core/tests/data',
+  'core/tests/load_tests',
+  'core/tests/release_sources',
+  'core/tests/services_sources',
+  'core/tests/test-dependencies',
+  'core/templates/tests',
+  'core/templates/services/UpgradedServices.ts',
+  'core/templates/services/angular-services.index.ts',
+  'core/templates/utility/hashes.ts',
+  'webpack.common.config.ts',
+  'webpack.common.macros.ts',
+  'webpack.dev.config.ts',
+  'webpack.dev.sourcemap.config.ts',
+  'webpack.prod.config.ts',
+  'webpack.prod.sourcemap.config.ts',
+  'angular-template-style-url-replacer.webpack-loader.js',
+];
+
+const FILE_EXTENSIONS = [
+  '.ts',
+  '.js',
+  '.html',
+  '.md',
+  '.css',
+  'CODEOWNERS',
+  'AUTHORS',
+  'CONTRIBUTORS',
+];
+
+const MANUALLY_MAPPED_DEPENDENCIES: Record<string, string[]> = {
+  '.lighthouserc-base.js': [
+    'puppeteer-login-script.js',
+    'core/tests/puppeteer/lighthouse_setup.js',
+  ],
+  'core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts': [
+    'core/tests/puppeteer-acceptance-tests/specs/helpers/reporter.ts',
+  ],
+  'core/templates/pages/header_css_libs.html': [
+    'core/templates/css/oppia.css',
+    'core/templates/css/oppia-material.css',
```

Comment:
I thought this will be done using code?

Reply:
Those were the niche cases where the imports were strange e.g. starting with /templates, think this is the only case and Sandeep mentioned that they will be removed after AngularJS migration.

Reply:
Makes sensee, lets keep it as is.